contentprovider.go

```diff
@@ -312,6 +316,16 @@ func (p *contentProvider) fillContentChunkMatches(mts matchTreeScorer, ms []*can
 	for _, chunk := range chunks {
 		score, debugScore, symbolInfo := p.candidateMatchScore(mts, chunk.candidates, language, debug)
 
+		if docScoreVec == nil {
```

Comment:
I had a different impression of how we would calculate this. What if we instead
* Scored each chunk by calculating the atom scores, then sum them together to get a single float
* Set the doc score equal to the max chunk score

This would reward documents that contain all terms on the same line, helping to boost docs with symbols like `SearchOptions`, over ones that contain the symbols `Search` and `Options` only completely different lines.

Reply:
Ok I will try this out. So that means we still assign a single float64 to each chunkmatch, and we take the max chunkmatch score to assign the document score? Atleast the part of the document score that comes from chunkmatches.